model.py

```diff
@@ -280,7 +289,216 @@ def _restrict_to_stages(cls, qu, allowed_stages=None):
         """
         allowed_stages = allowed_stages or [Library.LIVE]
         return qu.filter(Library.stage.in_(allowed_stages))
-    
+
+    @classmethod
+    def relevant(cls, _db, target, language, audiences=None, allowed_stages=None):
+        """Find libraries that are most relevant for a user.
+        
+        :param target: The user's current location. May be a Geometry object or
+        a 2-tuple (latitude, longitude).
+        :param language: The ISO 639-1 code for the user's language.
+        :param audiences: List of audiences the user is a member of.
+        By default, only libraries with the PUBLIC audience are shown.
+        :param allowed_stages: A list of allowable values for
+        `Library.stage`. By default, only libraries in the LIVE stage
+        are shown.
+
+        :return A list of (Library, score) tuples, sorted by relevance.
```

Comment:
I think leave it like this. We can put the score in the feed and use it to troubleshoot problems.